### PR TITLE
Fix n3h version check

### PR DESCRIPTION
**Pull request by lucksus**
## PR summary

Although I made the pinned n3h version match the version in the n3h branch I'm working on, I couldn't get the conductor to use my local n3h from $PATH. Found out that the version check fails if the string returned from the spawned command does not match the regex, although in my case the string was exactly the version that was pinned.

This adds a plain `res == &N3H.version` test and refactors that function so the flow easier to understand.

- [ ] this is a code change that effects some consumer (e.g. zome developers) of holochain core so there is an 'Unreleased' CHANGELOG item, with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [x] this is not a code change, or doesn't effect anyone outside holochain core development

### CODE REVIEW

File: net/src/ipc/n3h.rs [hunk 1]
```diff
@@ -150,27 +150,33 @@ fn check_n3h_version(path: &std::path::PathBuf) -> NetResult<Vec<String>> {
...
-        let res = res.unwrap();
+    let res = exec_output(path, out_args, ".", false).map_err(|err| {
+        println!("response: {:?}", err);
+        format_err!("n3h not found in path: {:?}", &path)
+    })?;
+
+    // `n3h --version` returns only the version number, for example "v0.0.11-alpha"
+    if res == N3H_INFO.version {
+        Ok(true)
+    } else {
```
Comment (ddd-mtl) [inline net/src/ipc/n3h.rs]: You can `return Ok(())` and remove the `else`.

File: net/src/ipc/n3h.rs [hunk 2]
```diff
@@ -150,27 +150,33 @@ fn check_n3h_version(path: &std::path::PathBuf) -> NetResult<Vec<String>> {
...
         let re = regex::Regex::new(r#"(?m)#\s+n3h\s+version:\s+"([^"]+)"\s+#$"#)?;
 
         let mut version = None;
         for m in re.captures_iter(&res) {
             version = Some(m[1].to_string());
         }
 
-        if version.is_none() || version.as_ref().unwrap() != &N3H_INFO.version {
+        if version.is_some() && version.as_ref().unwrap() == &N3H_INFO.version {
+            Ok(true)
```
Comment (ddd-mtl) [inline net/src/ipc/n3h.rs]: same here

File: net/src/ipc/n3h.rs [hunk 3]
```diff
@@ -150,27 +150,33 @@ fn check_n3h_version(path: &std::path::PathBuf) -> NetResult<Vec<String>> {
 }
 
 fn sub_check_n3h_version(path: &std::path::PathBuf, out_args: &[&str]) -> NetResult<bool> {
-    let res = exec_output(path, out_args, ".", false);
```
Comment (ddd-mtl) [inline net/src/ipc/n3h.rs]: Might as well return `NetResult<()>`, that boolean is not used for anything.

**Comment by lucksus** [comment]
Yes, @neonphog, I see this:
```
lucksus@terence:~$ n3h --version
v0.0.11-alpha
lucksus@terence:~$ 
```
That is from https://github.com/holochain/n3h/pull/71 in which I had merged develop when I worked on this last Thursday.

**Comment by zippy** [comment]
Is this ready to land @neonphog @lucksus ?

**Comment by lucksus** [comment]
Yes, @neonphog, you're correct. With n3h's master in my n3h branch I get the right version output, so yeah, this is not needed. Will close this PR.

**Pull request closed without merging**